include/ddsrouter/participant/implementations/rtps/impl/DiscoveryServerParticipant.ipp

```diff
@@ -353,32 +338,92 @@ void DiscoveryServerParticipant<ConfigurationType>::enable_tls(
     descriptor->tls_config.add_option(
         eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSOptions::NO_SSLV2); // not safe
 
-    // CA certificate
-    descriptor->tls_config.verify_file = get_param(tls_config, TLS_CA_TAG);
     // Perform verification of the server
     descriptor->tls_config.add_verify_mode(
         eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSVerifyMode::VERIFY_PEER);
 
-    if (client_only)
+    if (client)
     {
-        // Fail verification if the server has no certificate
-        descriptor->tls_config.add_verify_mode(
-            eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSVerifyMode::VERIFY_FAIL_IF_NO_PEER_CERT);
+        if (!tls_configuration->can_be_client())
+        {
+            logError(DDSROUTER_DISCOVERYSERVER_PARTICIPANT,
+                    "TLS Configuration expected a Client configuration.");
+            throw ConfigurationException("TLS Configuration expected a Client configuration.");
```

Comment:
Should be `enable_tls` have the `noexcept` property? Or handle this exception somewhere else?

Reply:
`participant_attributes_()` could throw exception, as TLS could throw exceptions. This should handle above.